# RAT 6.3  

We will use the data in `https://s3-us-west-2.amazonaws.com/dsci/6002/data/homer.csv`.  

1) Fit a multiple regression model to predict $y$.  

In [1]:
%pylab inline

import pandas as pd
import seaborn as sns
import statsmodels.api as sm

Populating the interactive namespace from numpy and matplotlib


In [2]:
df = pd.read_csv("https://s3-us-west-2.amazonaws.com/dsci/6002/data/homer.csv")

In [3]:
df.head()

y        x1        x2        x3        x4        x5       x6
0 -2.17360 -0.023816 -0.012885  0.045936 -0.182890  0.028694 -0.81119
1 -1.29570  0.018453  0.058999 -0.072399  0.008679 -0.084344 -0.24208
2  0.14539  0.054999  0.131770 -0.067223  0.202430  0.127060  0.63388
3  0.92959 -0.018933 -0.010875 -0.028255  0.017912  0.084330  0.66062
4  0.90259 -0.000877  0.066941 -0.048190  0.064742  0.147880  0.26643

In [8]:
df = df[df['y'] > 0]

In [9]:
X = df[['x1', 'x2', 'x3', 'x4', 'x5','x6']]
X = sm.add_constant(X)
y = df['y']

model = sm.OLS(y, X)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.063
Model:                            OLS   Adj. R-squared:                  0.061
Method:                 Least Squares   F-statistic:                     32.31
Date:                Thu, 06 Oct 2016   Prob (F-statistic):           7.66e-38
Time:                        13:45:35   Log-Likelihood:                -2922.2
No. Observations:                2902   AIC:                             5858.
Df Residuals:                    2895   BIC:                             5900.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const          0.9325      0.014     67.183      0.000         0.905     0.960
x1             0.7076      0.494      1.432      0.152        -0.261     1.677
x2             0.5419      0.244      2.219      0.027         0.063     1.021
x3             0.8036      0.344      2.338      0.019         0.130     1.478
x4             0.2060      0.122      1.695      0.090        -0.032     0.444
x5             1.0507      0.166      6.311      0.000         0.724     1.377
x6             0.2959      0.023     12.842      0.000         0.251     0.341
==============================================================================
Omnibus:                      133.701   Durbin-Watson:                   2.011
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               65.134
Skew:                           0.168   Prob(JB):                     7.18e-15
Kurtosis:                       2.347   Cond. No.                         44.3
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

2) Comment on the fit.  

F-stat: 32.31 (not too high).

The overall p-value is very low: 7.66e-38

The adjusted r squared is very low: 0.061.

3) Interpret the slope coefficient of $x_1$.  

For every unit of increase of x1, y increases by 0.7076.

4) Is the model useful for predicting $y$? Why or why not?

I would remove variables x1 and x4 due to it's high p-value (higher than 0.05).

The adjusted r squared is very low: 0.061. Even though the overall p-value is very low (7.66e-38), I don't think it's a good model.

In [10]:
X = df[['x2', 'x3','x5','x6']]
X = sm.add_constant(X)
y = df['y']

model = sm.OLS(y, X)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.061
Model:                            OLS   Adj. R-squared:                  0.060
Method:                 Least Squares   F-statistic:                     47.16
Date:                Thu, 06 Oct 2016   Prob (F-statistic):           1.85e-38
Time:                        13:49:45   Log-Likelihood:                -2924.7
No. Observations:                2902   AIC:                             5859.
Df Residuals:                    2897   BIC:                             5889.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const          0.9366      0.014     68.056      0.000         0.910     0.964
x2             0.7226      0.211      3.424      0.001         0.309     1.136
x3             0.5550      0.315      1.762      0.078        -0.063     1.173
x5             1.1976      0.146      8.187      0.000         0.911     1.484
x6             0.3030      0.023     13.275      0.000         0.258     0.348
==============================================================================
Omnibus:                      138.508   Durbin-Watson:                   2.013
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               68.081
Skew:                           0.181   Prob(JB):                     1.65e-15
Kurtosis:                       2.343   Cond. No.                         30.4
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

By removing x1 and x4, the model improves slightly (higher F-statistic, lower overal p-value). I would exclude x3 dur to high p-value.

In [11]:
X = df[['x2','x5','x6']]
X = sm.add_constant(X)
y = df['y']

model = sm.OLS(y, X)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.060
Model:                            OLS   Adj. R-squared:                  0.059
Method:                 Least Squares   F-statistic:                     61.80
Date:                Thu, 06 Oct 2016   Prob (F-statistic):           1.00e-38
Time:                        13:51:27   Log-Likelihood:                -2926.3
No. Observations:                2902   AIC:                             5861.
Df Residuals:                    2898   BIC:                             5884.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const          0.9389      0.014     68.493      0.000         0.912     0.966
x2             0.4801      0.160      2.999      0.003         0.166     0.794
x5             1.1470      0.143      7.994      0.000         0.866     1.428
x6             0.3062      0.023     13.459      0.000         0.262     0.351
==============================================================================
Omnibus:                      143.948   Durbin-Watson:                   2.012
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               70.182
Skew:                           0.186   Prob(JB):                     5.76e-16
Kurtosis:                       2.336   Cond. No.                         14.5
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The above model is even better and the p-values of all independent variables is less than 0.05, so they are all significant (x2, x5 and x6).